### PR TITLE
i723: Added Quick Judge Pane.   Fixes 723

### LINKED ISSUE #723: Provide a way to rejudge and/or update judgement en masse
**Is your feature request related to a problem?** 

There is a need to rejudge en masse and update judgements for runs en masse

User requested:
   Rejudge selected runs
   Rejudge all solution of a certain problem
   Change verdicts of all selected runs

**Feature Description**: 

On the admin, provide a way to list runs, multi select runs and:
rejudge all selected runs
update all selected runs judgements 
   for debugging purposes add an ability to update judgements
Provide a Filter button (like the Runs tab)
Provide a number of runs display (like the Runs tab)
Provide a View Unjudged only checkbox

before rejudging or updating judgement prompt Are you sure?
  with Yes/No/Cancel  (and the number of runs affected and a description of the change to the runs)

for a login to rejudge must have JUDGE_RUN Judge runs permission
for a login to update must have EDIT_RUN Edit a run permission

Add to UI Plugins pane

**Have you considered other ways to accomplish the same thing?** 

Requires code change

**Do you have any specific suggestions for how your feature would be ***implemented*** in PC^2?**

Add a new Pane under Run Contest named TBD.

**Additional context**: 

If a team uses this tab, the only runs they can view are their runs
they cannot update their run's judgements
they cannot rejudge their runs

### PR DESCRIPTION
**Pull request by lane55**
### Description of what the PR does

Adds Quick Judge Pane on Admin under Run Contests
Added to UI Plugin list

### Issue which the PR addresses

Fixes #723 

### Environment in which the PR was developed (OS,IDE, Java version, etc.)

Microsoft Windows [Version 10.0.19044.2130]

java version "1.8.0_121"
Java(TM) SE Runtime Environment (build 1.8.0_121-b13)
Java HotSpot(TM) 64-Bit Server VM (build 25.121-b13, mixed mode)

### Precise steps for _testing_ the PR (i.e., how to demonstrate that it works correctly)

See #723 for Pane components

Setup 
load and start server, --load tenprobs (contest time starts automatically, with 80 teams)
submit 12 runs as different teams

Test 1 - Change 6 runs to Yes

Note the run count in the top right corner, it will
   change as runs are judged/"rejudged"/filtered

Run Contest -> Quick Judge 
 - select 6 runs
 - change judgement (pull down) to Yes
 - click Submit
 - click Yes

View judged runs
 - uncheck View Unjudged only, 
   see run count 

 - check View Unjudged only, 
   confirm run count change with fewer runs

Rejudge 3 runs
 - uncheck View Unjudged only, see run count 
 - select 3 runs
 - click rejudge
 - click Yes 

 test: those 3 runs are queued or manual run state (depending on judging type)

Judge all runs 
 - select all runs
 - change Judgement to No Output Format Error
 - click Submit
 - click Yes

  test: Yes runs do not change judgement
  test: the queued runs are now No Output Format Error

 To change already judged runs,
 - Rejudge the runs
 test: runs change status to Queued
 - change Judgement to No Output Format Error
 - click Submit
 - click Yes
 test: selected runs judgement changed to No Output Format Error

To test team permissions (buttons not enabled)
 - start pc2rui with a team number (that has submitted runs)
    bin\pc2rui --login 5
 - select UIPlugins
 - pick Quick Judge Panel
 - click Add Tab
 - select Quick Judge Panel
 (may need to Uncheck View Unjuded Only to see runs)
 Test: Rejudged button not enabled
 Test: Submit button not enabled

### CODE REVIEW

File: src/edu/csus/ecs/pc2/ui/admin/AdministratorView.java [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Copyright (C) 1989-2019 PC2 Development Team: John Clevenger, Douglas Lane, Samir Ashoo, and Troy Boudreau.
+// Copyright (C) 1989-2013 PC2 Development Team: John Clevenger, Douglas Lane, Samir Ashoo, and Troy Boudreau.
```
Comment (johnbrvc) [inline src/edu/csus/ecs/pc2/ui/admin/AdministratorView.java:1]: Copyright date should be 2023.

File: src/edu/csus/ecs/pc2/ui/QuickJudgePane.java
Comment (johnbrvc) [inline src/edu/csus/ecs/pc2/ui/QuickJudgePane.java:1]: You asked for alternate names for this class.  Possible suggestions:
Judge Utility
Judge Tool

File: src/edu/csus/ecs/pc2/ui/QuickJudgePane.java [hunk 2]
```diff
@@ -0,0 +1,893 @@
...
+    /**
+     * This method initializes judgementComboBox
+     * 
+     * @return javax.swing.JComboBox
+     */
+    private JComboBox<Judgement> getJudgementComboBox() {
+        if (judgementComboBox == null) {
+            judgementComboBox = new JComboBox<Judgement>();
+            judgementComboBox.setPreferredSize(new java.awt.Dimension(201, 25));
+            judgementComboBox.setPreferredSize(new java.awt.Dimension(201, 25));
```
Comment (johnbrvc) [inline src/edu/csus/ecs/pc2/ui/QuickJudgePane.java:559]: Why are there 2 of these statements?

File: src/edu/csus/ecs/pc2/ui/QuickJudgePane.java [hunk 3]
```diff
@@ -0,0 +1,893 @@
...
+                slog("Quick Judge run " + id + " with judgement '" + judgement + "' ");
+
+                boolean judgedAsSolved = judgementIndex == 2;
+
+                Run run = getContest().getRun(id);
+
+                // Create judgement and save it in pendingJudgements
+                JudgementRecord judgementRecord = new JudgementRecord(judgement.getElementId(), getContest().getClientId(), judgedAsSolved, false);
+                pendingJudgements.put(run.getElementId(), judgementRecord);
+                System.out.println("Stored/Pending " + run);
```
Comment (johnbrvc) [inline src/edu/csus/ecs/pc2/ui/QuickJudgePane.java:629]: Probably want to remove System.out.xxxx or bracket it with a test for debugging or some such.

Comment (lane55) [reply src/edu/csus/ecs/pc2/ui/QuickJudgePane.java:629]: Will replace System.out with log statements.

Comment (lane55) [reply src/edu/csus/ecs/pc2/ui/QuickJudgePane.java]: pushed fix.

File: src/edu/csus/ecs/pc2/ui/admin/AdministratorView.java (hunk 1 above)
Comment (lane55) [reply src/edu/csus/ecs/pc2/ui/admin/AdministratorView.java]: pushed fix

File: src/edu/csus/ecs/pc2/ui/QuickJudgePane.java
Comment (lane55) [reply src/edu/csus/ecs/pc2/ui/QuickJudgePane.java:1]: changed tab name to "Judging Utilities"
pushed fix.

**Review by johnbrvc — APPROVED** [review]
Works as I would expect.  I am definitely not in love with the logging stuff; slog someplaces, log.info others, etc, but there is a comment in the code to refactor someday.

**Review by troy2914 — APPROVED** [review]
Seems reasonable to me,  did some light testing on it.

**Pull request merged**